lib/io/can.hpp

```diff
@@ -23,10 +27,18 @@ struct CanFrame {
 
 enum class CanResult { kFailure, kSuccess };
 
+class ICanProcessor {
+ public:
+  virtual void processMessage(const io::CanFrame frame);
+};
+
 class ICan {
+ public:
   virtual CanResult initialise(const std::string &can_network_interface);
   virtual CanResult send(const CanFrame &message);
   virtual std::optional<CanFrame> receive();
+  virtual CanResult addCanProcessor(uint16_t ID, ICanProcessor processor);
+  virtual CanResult removeCanProcessor(ICanProcessor processor);
```

Comment:
all pure virtual, please

Reply:
Also `removeCanProcessor` may be hard to realise.

1. it's not something we anticipate to need
2. removing things from a vector is hard so we would require a different data structure (`unordered_set`)

I'd say scratch this.